tests/unit_tests/test_wandb_artifacts_full.py

```diff
@@ -193,3 +194,29 @@ def make_table():
     artifact2.add(t1, "t2")
     assert artifact2.manifest.entries["t2.table.json"].ref is not None
     run.finish()
+
+
+def test_artifact_wait_success(wandb_init):
+    # Test artifact wait() timeout parameter
+    timeout = 2
+    leeway = 0.50
+    run = wandb_init()
+    artifact = wandb.Artifact("art", type="dataset")
+    start_timestamp = time.time()
+    run.log_artifact(artifact).wait(wait_timeout_secs=timeout)
+    elapsed_time = time.time() - start_timestamp
+    assert elapsed_time < timeout * (1 + leeway)
+    run.finish()
+
+
+@pytest.mark.parametrize("timeout", [0, 1e-1])
```

Comment:
IIUC this timeout is supposed to be "small enough that the artifact-upload will never complete in less than this many seconds"; but 0.1sec is long enough that it's within the range of plausible execution times. If there's no countervailing reason to keep it large, could we make it Obviously Incredibly Small, like 1e-6, or 1e-12?

(I'm partly worried about hiccups in CI scheduling causing this test to flake, and partly just trying to make the code as obvious as possible)